provider/pkg/provider/team.go

```diff
@@ -195,28 +194,33 @@ func (tr *PulumiServiceTeamResource) Create(req *pulumirpc.CreateRequest) (*pulu
 		return nil, err
 	}
 
-	inputsTeam := tr.ToPulumiServiceTeamInput(inputs)
+	inputsTeam := ToPulumiServiceTeamInput(inputs)
 	team, err := tr.createTeam(ctx, inputsTeam)
 	if err != nil {
 		return nil, fmt.Errorf("error creating team '%s': %s", inputsTeam.Name, err.Error())
 	}
 
+	// We have now created a team.  It is very important to ensure that from this point on, any other error
+	// below returns the ID using the `pulumirpc.ErrorResourceInitFailed` error details annotation.  Otherwise,
+	// we leak a team resource. We ensure that we wrap any errors in a partial error and return that to the RPC.
+
+	// make copy of input so we can safely modify output without affecting input
+	inProgTeam := ToPulumiServiceTeamInput(inputsTeam.ToPropertyMap())
+	inProgTeam.Members = nil
+
 	for _, memberToAdd := range inputsTeam.Members {
-		err = tr.addToTeam(ctx, inputsTeam.OrganizationName, inputsTeam.Name, memberToAdd)
+		err := tr.addToTeam(ctx, inputsTeam.OrganizationName, inputsTeam.Name, memberToAdd)
 		if err != nil {
-			return nil, err
+			return nil, partialError(*team, err, inProgTeam, inputsTeam)
 		}
+		// if we've successfully added member to team, save them to the state we're going to return
+		// so that a re-run can detect the left over members to add via Update
+		inProgTeam.Members = append(inProgTeam.Members, memberToAdd)
 	}
 
-	outputStore := resource.PropertyMap{}
-	outputStore["__inputs"] = resource.NewObjectProperty(inputs)
```

Comment:
For the sake of simplicity, I lifted the input properties from `__inputs` to the top level in the output. 

Initially, I was concerned that lifting this would conflict with resources that were created with an older version of the provider. However, this didn't seem to be an issue, as inside `Update` & `Diff` we reference `inputs` directly.

I tested this by creating a team with the old version of the provider, then upgrading the provider, and then running another update. Diffs and updates worked as expected.